File: backend/Domain/Message.cs
namespace backend.Domain
{
    public class Message
    {
        public Guid Id { get; set; }
        public Guid SenderId { get; set; }
        public User Sender { get; set; } = null!;
        public Guid ReceiverId { get; set; }
        public User Receiver { get; set; } = null!;
        public string Content { get; set; } = string.Empty;
        public DateTime SentAt { get; set; } = DateTime.UtcNow;
        public bool IsRead { get; set; } = false;
    }
}

File: frontend/domain/def/Message.ts
import { EMPTY_GUID, type TNullable } from "@/types";
import { IMessage } from "../meta/IMessage";

class Message implements IMessage {
  senderId: string;
  receiverId: string;
  content: TNullable<string>;
  sentAt: Date;
  isRead: boolean;
  constructor(
    senderId = EMPTY_GUID,
    receiverId = EMPTY_GUID,
    content: TNullable<string> = null,
    sentAt = new Date(),
    isRead = false,
  ) {
    this.senderId = senderId;
    this.receiverId = receiverId;
    this.content = content;
    this.sentAt = sentAt;
    this.isRead = isRead;
  }
}

export { Message };
